client/src/components/History/CurrentHistory/HistoryFilters.test.js

```diff
@@ -0,0 +1,53 @@
+import { mount } from "@vue/test-utils";
+import { getLocalVue } from "jest/helpers";
+import HistoryFilters from "./HistoryFilters";
+const localVue = getLocalVue();
+
+describe("HistoryFilters", () => {
+    it("test history filter panel", async () => {
+        const wrapper = mount(HistoryFilters, {
+            propsData: {
+                filterText: "",
+                showAdvanced: false,
+            },
+            localVue,
+            stubs: {
+                icon: { template: "<div></div> " },
+            },
+        });
+        expect(wrapper.findAll("[description='advanced filters']").length).toBe(0);
+        await wrapper.setProps({ showAdvanced: true });
+        expect(wrapper.findAll("[description='advanced filters']").length).toBe(1);
+        const filterInputs = {
+            "[placeholder='any name']": "name-filter",
+            "[placeholder='any extension']": "ext-filter",
+            "[placeholder='any tag']": "tag filter",
+            "[placeholder='any state']": "state-filter",
+            "[placeholder='index greater']": "hid-greater",
+            "[placeholder='index lower']": "hid-lower",
+            "[placeholder='created after']": "January 1, 2022",
+            "[placeholder='created before']": "January 1, 2023",
+        };
+        Object.entries(filterInputs).forEach(([selector, value]) => {
+            const filterInput = wrapper.find(selector);
+            if (filterInput.vm && filterInput.props().type == "text") {
+                filterInput.setValue(value);
+            }
+        });
+        const searchButton = wrapper.findAll("[description='apply filters']");
+        searchButton.trigger("click");
```

Comment:
```suggestion
        await searchButton.trigger("click");
```
Maybe not critical here, but it's the recommended way of calling trigger in an async context, just for consistency :)